### PR TITLE
FIX: Modifications to GridFieldExportButton to allow ArrayList use in SS_Report

### PR DESCRIPTION
**Pull request by wilr**
- canView is not defined, so if method doesn't exist assume that the user can view it (what GridField does).

- check destroy method exists

- Get value by source number returned null on arraylist, this will fetch by column name

### CODE REVIEW

**Comment by wilr** [comment]
@tractorcow Probably should go against the 3 branch rather than 3.1 but figured as a bug fix may get in. Can rebase it if it's not suitable.

File: forms/gridfield/GridFieldExportButton.php [hunk 1]
```diff
@@ -132,7 +132,7 @@ public function generateExportFileData($gridField) {
 		}
 
 		foreach($items->limit(null) as $item) {
-			if($item->hasMethod('canView') && $item->canView()) {
+			if(!$item->hasMethod('canView') || $item->canView()) {
```
Comment (tractorcow) [inline forms/gridfield/GridFieldExportButton.php]: `$item->hasMethod('hasMethod')` ? :P

Comment (wilr) [inline forms/gridfield/GridFieldExportButton.php]: CanView?

Comment (tractorcow) [inline forms/gridfield/GridFieldExportButton.php]: I was making a useless joke sorry. It was a "coffee machine is broken" week.

**Comment by tractorcow** [comment]
And I forgot to merge it as well. Yikes sorry @wilr

**Pull request merged**